jobs/build/ocp/Jenkinsfile

```diff
@@ -642,11 +655,68 @@ Please direct any questsions to the Continuous Delivery team (#aos-cd-team on IR
                 sh "ose_images.sh --user ocp-build build_container --branch rhaos-${BUILD_VERSION}-rhel-7 --group base --repo https://raw.githubusercontent.com/openshift/aos-cd-jobs/master/build-scripts/repo-conf/aos-unsigned-building.repo"
             }
 
-            buildlib.oit """
+            waitUntil {
+              try {
+                exclude = ""
+                if ( BUILD_EXCLUSIONS != "" ) {
+                  exclude = "-x ${BUILD_EXCLUSIONS} --ignore-missing-base"
+                }
+                buildlib.oit """
 --working-dir ${OIT_WORKING} --group openshift-${BUILD_VERSION}
+${exclude}
 images:build
 --push-to-defaults --repo-type unsigned
 """
+                return true // finish waitUntil
+              }
+              catch ( err ) {
+                try {
+                  record_log = buildlib.parse_record_log( OIT_WORKING )
+                  builds = record_log['build']
+                  failed_builds = "FAILED BUILDS"
+                  for(i = 0; i < builds.size(); i++) {
+                    if ( builds[i]['status'] == '-1' ){
+                      distgit = build[i]['distgit']
+                      task_url = "No Brew task available, see jenkins log."
+                      if ( build[i].containsKey('task_url') ) {
+                        task_url = build[i]['task_url']
+                      }
+                    }
+                    
+                    failed_builds += "${distgit}: ${task_url}\n"
+                  }
+                  mail(to: "${MAIL_LIST_FAILURE}",
+                        from: "aos-cd@redhat.com",
+                        subject: "RESUMABLE Error during Image Build for OCP v${BUILD_VERSION}",
+                        body: """Encountered an error: ${err}
+Input URL: ${env.BUILD_URL}input
+Jenkins job: ${env.BUILD_URL}
+
+FAILED BUILDS:
+${failed_builds}
+""");
+                } catch ( mail_exception ) {
+                    echo "One or more errors sending email: ${mail_exception}"
+                }
+                
+                def resp = input    message: "Error during Image Build for OCP v${BUILD_VERSION} with args: ${opts}",
+                                  parameters: [
+                                                  [$class: 'hudson.model.ChoiceParameterDefinition',
+                                                   choices: "RETRY\nCONTINUE\nABORT",
+                                                   description: 'Retry (try the operation again). Continue (fails are OK, continue pipeline). Abort (terminate the pipeline).',
+                                                   name: 'action']
+                                  ]
+
+                if ( resp == "RETRY" ) {
+                    return false  // cause waitUntil to loop again
+                } else if ( resp == "CONTINUE" ) {
+                    echo "User chose to build fails are OK."
```

Comment:
This codepath somehow needs to affect the PARTIAL build email log. We know images failed and so should consumers of that build.